### ISSUE TITLE
wt build failure

### ISSUE THREAD

**Issue by FrankHeimes** [microsoft/vcpkg#9859]
**Host Environment**
- OS: Windows 10 Enterprise 1903
- Compiler: Visual Studio 2019, 16.4.3, Toolset v142

**To Reproduce**
* On 25 January 2020, I updated my entire vcpkg folder from github and rebuilt vcpkg.
* `.\vcpkg remove --outdated --recurse`
* `.\vcpkg update`
* `.\vcpkg install P` for P = `boost`, ..., `wt`

This worked fine for about 200 libraries, except for `wt`

The command  `./vcpkg install wt`  displays the following list and immediately returns to the command line. Note that `wt[core]:x86-windows` is the only package left to build but vcpkg does not even attempt to build it.
```
The following packages will be rebuilt:
  * harfbuzz[core,glib,ucdn]:x86-windows
  * qt5[core]:x86-windows
  * qt5-3d[core]:x86-windows
  * qt5-activeqt[core]:x86-windows
  * qt5-base[core]:x86-windows
  * qt5-charts[core]:x86-windows
  * qt5-datavis3d[core]:x86-windows
  * qt5-declarative[core]:x86-windows
  * qt5-gamepad[core]:x86-windows
  * qt5-graphicaleffects[core]:x86-windows
  * qt5-imageformats[core]:x86-windows
  * qt5-location[core]:x86-windows
  * qt5-mqtt[core]:x86-windows
  * qt5-multimedia[core]:x86-windows
  * qt5-networkauth[core]:x86-windows
  * qt5-purchasing[core]:x86-windows
  * qt5-quickcontrols[core]:x86-windows
  * qt5-quickcontrols2[core]:x86-windows
  * qt5-script[core]:x86-windows
  * qt5-scxml[core]:x86-windows
  * qt5-sensors[core]:x86-windows
  * qt5-serialport[core]:x86-windows
  * qt5-speech[core]:x86-windows
  * qt5-svg[core]:x86-windows
  * qt5-tools[core]:x86-windows
  * qt5-virtualkeyboard[core]:x86-windows
  * qt5-webchannel[core]:x86-windows
  * qt5-websockets[core]:x86-windows
  * qt5-winextras[core]:x86-windows
  * qt5-xmlpatterns[core]:x86-windows
The following packages will be built and installed:
  * boost-signals[core]:x86-windows
  * cairo[core]:x86-windows
  * dirent[core]:x86-windows
  * expat[core]:x86-windows
  * fontconfig[core]:x86-windows
  * gettext[core]:x86-windows
  * glew[core]:x86-windows
  * glib[core]:x86-windows
  * libiconv[core]:x86-windows
  * pango[core]:x86-windows
  * pcre[core]:x86-windows
  * pixman[core]:x86-windows
    wt[core]:x86-windows
Additional packages (*) will be modified to complete this operation.
If you are sure you want to rebuild the above packages, run the command with the --recurse option
```

**Comment by PhoebeHui** [comment]
@FrankHeimes, It seems that you need  '.\vcpkg upgrade --no-dry-run' after executing '.\vcpkg update'.

**Comment by PhoebeHui** [comment]
BTW, I tried the steps, and looks everything works well for me.
Starting package 102/102: wt:x86-windows
Building package wt[core]:x86-windows...
-- Downloading https://github.com/emweb/wt/archive/2441ac6dc7a208a33aaa06f62f065f8567bf94f3.tar.gz...
-- Extracting source E:/VCPKG/src/downloads/emweb-wt-2441ac6dc7a208a33aaa06f62f065f8567bf94f3.tar.gz
-- Applying patch 0002-link-glew.patch
-- Applying patch 0003-disable-boost-autolink.patch
-- Applying patch 0004-link-ssl.patch
-- Applying patch 0005-XML_file_path.patch
-- Using source at E:/VCPKG/src/buildtrees/wt/src/8567bf94f3-b84c4b8425
-- Configuring x86-windows
-- Building x86-windows-dbg
-- Building x86-windows-rel
-- Installing: E:/VCPKG/src/packages/wt_x86-windows/share/wt/copyright
-- Performing post-build validation
-- Performing post-build validation done
Building package wt[core]:x86-windows... done
Installing package wt[core]:x86-windows...
Installing package wt[core]:x86-windows... done
Elapsed time for package wt:x86-windows: 8.934 min

Total elapsed time: 1.021 h

The package wt:x86-windows provides CMake targets:

    find_package(wt CONFIG REQUIRED)
    # Note: 3 target(s) were omitted.
    target_link_libraries(main PRIVATE Wt::Wt Wt::Dbo Wt::HTTP Wt::Isapi)

**Comment by FrankHeimes** [comment]
> @FrankHeimes, It seems that you need '.\vcpkg upgrade --no-dry-run' after executing '.\vcpkg update'.

I also tried that. Same behavior.

**Comment by PhoebeHui** [comment]
@FrankHeimes, could you get vcpkg latest source and try again, I tried several times, still not repro this issue.

I tried the steps:
git pull
.\bootstrap-vcpkg.bat
.\vcpkg remove --outdated --recurse
.\vcpkg update
.\vcpkg upgrade --no-dry-run
.\vcpkg install wt

**Comment by FrankHeimes** [comment]
I had set `VCPKG_LIBRARY_LINKAGE` to `static` in the triplets and therefore had an issue with `glib` enforcing **all** dependent `vcpkg` packages to create dynamic libraries only. I was able to fix that.

What irritated me was that `vcpkg` attempted to build the dependent package `glib` when I tried to install `cairo`, which `wt` depends on. But `vcpkg` did not attempt to build any dependent packages automatically when I tried to install `pango` or `wt`.

The lesson here is that you always have to specify `vcpkg install --recurse` in cases like this.
